packages/profile-sync-controller/src/controllers/user-storage/__fixtures__/test-utils.ts

```diff
@@ -0,0 +1,35 @@
+type WaitForOptions = {
+  intervalMs?: number;
+  timeoutMs?: number;
+};
+
+/**
+ * Testing Utility - waitFor. Waits for and checks (at an interval) if assertion is reached.
+ *
+ * @param assertionFn - assertion function
+ * @param options - set wait for options
+ * @returns promise that you need to await in tests
+ */
+export const waitFor = async (
```

Comment:
Test util for waiting for some things to finish - e.g. when testing when a publish event has been listened to and completed.

This is similar interface to react-testing-library. LMK if there is an existing package we can utilise instead of re-inventing the wheel.

Reply:
I've been having the same issue and decided to go with monitoring another synchronous function I could mock. Will definitely be useful!